eclipse-theia/theia
# Capture stopwatch results

#### What it does

Introduce `MeasurementResults` that capture a stopwatch execution in a serializable format. 
Add `onMeasurementResult` event emitter to stopwatch class and introduce optional  caching of results.
Add a metrics contribution to `@theia/metrics` that collects all measurement results of frontend and backend processes (when running with debug log level).

The new metrics can be picked up by the  performance measurement step of the [e2e framework](https://github.com/eclipse-theia/theia-e2e-test-suite/pull/9).

Contributed on behalf of STMicroelectronics

#### How to test

Start the example Theia application in browser in debug mode (`yarn browser start:debug`). Go to `localhost:3000/metrics`. The stopwatch results for the backend and frontend should be part of the dataset.
Reload the theia application, then reload `localhost:3000/metrics`. It should now contain an additional dataset for the second frontend instance.
#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
